lib/pry/helpers/base_helpers.rb

```diff
@@ -1,138 +1,71 @@
-class Pry
-  module Helpers
-
-    module BaseHelpers
-
-      module_function
-
-      def silence_warnings
-        old_verbose = $VERBOSE
-        $VERBOSE = nil
-        begin
-          yield
-        ensure
-          $VERBOSE = old_verbose
-        end
-      end
-
-      # Acts like send but ignores any methods defined below Object or Class in the
-      # inheritance hierarchy.
-      # This is required to introspect methods on objects like Net::HTTP::Get that
-      # have overridden the `method` method.
-      def safe_send(obj, method, *args, &block)
-        (Module === obj ? Module : Object).instance_method(method).bind(obj).call(*args, &block)
-      end
-      public :safe_send
-
-      def find_command(name, set = Pry::Commands)
-        command_match = set.find do |_, command|
-          (listing = command.options[:listing]) == name && listing != nil
-        end
-        command_match.last if command_match
-      end
-
-      def not_a_real_file?(file)
-        file =~ /^(\(.*\))$|^<.*>$/ || file =~ /__unknown__/ || file == "" || file == "-e"
-      end
-
-      def command_dependencies_met?(options)
-        return true if !options[:requires_gem]
-        Array(options[:requires_gem]).all? do |g|
-          Rubygem.installed?(g)
-        end
-      end
-
-      def use_ansi_codes?
-        windows_ansi? || ENV['TERM'] && ENV['TERM'] != "dumb"
-      end
-
-      def colorize_code(code)
-        CodeRay.scan(code, :ruby).term
-      end
-
-      def highlight(string, regexp, highlight_color=:bright_yellow)
-        string.gsub(regexp) { |match| "<#{highlight_color}>#{match}</#{highlight_color}>" }
-      end
-
-      # formatting
-      def heading(text)
-        text = "#{text}\n--"
-        "\e[1m#{text}\e[0m"
-      end
-
-      #
-      # @return [Boolean]
-      #  Returns true if Pry is running on Mac OSX.
-      #
-      # @note
-      #   Queries RbConfig::CONFIG['host_os'] with a best guess.
-      #
-      def mac_osx?
-        !!(RbConfig::CONFIG['host_os'] =~ /\Adarwin/i)
-      end
-
-      #
-      # @return [Boolean]
-      #   Returns true if Pry is running on Linux.
-      #
-      # @note
-      #   Queries RbConfig::CONFIG['host_os'] with a best guess.
-      #
-      #
-      def linux?
-        !!(RbConfig::CONFIG['host_os'] =~ /linux/i)
-      end
+module Pry::Helpers; end
+module Pry::Helpers::BaseHelpers
```

Comment:
why not just nest it:

```ruby
module Pry::Helpers
  module Base
     ...
  end
end
```

Reply:
I prefer opening classes and modules like so:
```ruby
class Foo::Bar::Baz
end
```
for the simple reason that you don't add unnecessary indentation levels, and constant lookup is absolute (Foo::Daz.new), so it is transparent to the reader. Leads to nicer looking code ime.